src/game/server/entities/scientist-mine.cpp

```diff
@@ -51,11 +51,11 @@ void CScientistMine::Explode(int DetonatedBy)
 	{
 		float Dist = distance(m_Pos, OwnerChar->m_Pos);
 		if(Dist < OwnerChar->m_ProximityRadius+g_Config.m_InfMineRadius)
-			OwnerChar->TakeDamage(vec2(0.0f, 0.0f), 4, DetonatedBy, WEAPON_HAMMER, TAKEDAMAGEMODE_NOINFECTION);
+			OwnerChar->TakeDamage(vec2(0.0f, 0.0f), 4, DetonatedBy, WEAPON_HAMMER, TAKEDAMAGEMODE_SELFHARM);
```

Comment:
why is this being changed in a single commit and in this pull request? It doesn't seem to have anything with turrets and heroes

Reply:
because of the new self harm code, humans that conflict damage that will hurt themselves, need to do so with TAKEDAMAGEMODE_SELFHARM, its the same as TAKEDAMAGEMODE_NOINFECTION but also does selfharm